kernel/kernel-api/src/main/java/io/delta/kernel/internal/metadatadomain/JsonMetadataDomain.java

```diff
@@ -100,9 +102,10 @@ protected static <T> T fromJsonConfiguration(String json, Class<T> clazz) {
    */
   protected static <T> Optional<T> fromSnapshot(
       SnapshotImpl snapshot, Class<T> clazz, String domainName) {
-    return Optional.ofNullable(snapshot.getDomainMetadataMap().get(domainName))
-        .filter(domainMetadata -> !domainMetadata.isRemoved())
-        .map(domainMetadata -> fromJsonConfiguration(domainMetadata.getConfiguration(), clazz));
+    Optional<DomainMetadata> domainMetadata =
+        Optional.ofNullable(snapshot.getDomainMetadataMap().get(domainName));
+    domainMetadata.ifPresent(dm -> checkArgument(!dm.isRemoved()));
```

Comment:
can this use the public method? we just use `getConfiguration` anyways, this seems like some code duplication  on `Snapshot::getDomainMetadata`